code/datums/status_effects/status_effect.dm

```diff
@@ -45,19 +46,22 @@
 		owner = null
 	return ..()
 
-/datum/status_effect/process()
+/datum/status_effect/process(delta_time)
 	if(!owner)
 		qdel(src)
 		return
 	if(tick_interval < world.time)
-		tick()
-		tick_interval = world.time + initial(tick_interval)
+		var/actual_tick_interval = initial(tick_interval)
+		var/tick_delta_time = (last_tick && actual_tick_interval) ? max((world.time - last_tick) / actual_tick_interval, 1) : 1
+		tick(tick_delta_time)
+		tick_interval = world.time + actual_tick_interval
+		last_tick = world.time
```

Comment:
Some comments describing these changes or the intended functionality of these changes would be appreciated. Right now its hard to tell if this is doing what is expected or not as I'm not sure what its meant to be doing in the first place.
I assume if the tick is faster than the subsystem can run, it will account for that?

Since you are adding in delta time, I'm going to have to ask that all the existing status effects be updated to properly account for it, since its a pretty important foundational change to status effects. How the delta_time functions within the context of tick() should be documented with a comment so that coders using it know what its actually doing without needing to test it and see if it does what they expect. delta_time in SS13 has many different uses and may refer to real time delta_time or tick proportional delta_time. Explain which one it is functioning as (I believe its based on real time rather than subsystem tick, which is different to how delta_time inside process works).

Reply:
> I'm not sure what its meant to be doing in the first place

It's meant to calculate a multiplier _relative to the tick_interval_, i.e "it's been 2x longer than it should have since this status effect last ticked"

Reply:
Could you explain that in a code comment?

Reply:
For the sake of atomization, I just went and entirely removed the delta_time code I added.